lib/smart_proxy_remote_execution_ssh/multiplexed_ssh_connection.rb

```diff
@@ -67,12 +69,18 @@ def initialize(options, logger:)
     def establish!
       @available_auth_methods ||= available_authentication_methods
       method = @available_auth_methods.find do |method|
-        if try_auth_method(method)
+        result, pm = try_auth_method(method)
+        method.errors = pm.stderr
+        if result
           @available_auth_methods.unshift(method).uniq!
           true
         end
```

Comment:
With the changes to `try_auth_method` this could be:

```suggestion
        result = try_auth_method(method)
        method.errors = result.stderr
        if result.status.zero?
          @available_auth_methods.unshift(method).uniq!
          true
        end
```